census/src/main/java/io/grpc/census/CensusStatsModule.java

```diff
@@ -706,4 +715,116 @@ public void onClose(Status status, Metadata trailers) {
+      this.fullMethodName = checkNotNull(fullMethodName);
+      this.stopwatch = module.stopwatchSupplier.get().start();
+    }
+
+    @Override
+    public CallAttemptTracer newCallAttemptTracer() {
+      callAttemptTracer = new CallAttemptStatsTracer(module, parentCtx, fullMethodName, stopwatch);
+      return callAttemptTracer;
+    }
+
+    void callEnded(Status status) {
+      if (callAttemptTracer.streamTracerFactories.size() == 1) {
+        // for the case retry not enable.
+        callAttemptTracer.attemptEnded(0, status);
+      }
+
+      stopwatch.stop();
+      MeasureLong RETRIES_PER_CALL =
+          Measure.MeasureLong.create(
+              "grpc.io/client/retries_per_call", "Number of retries per call", "1");
+      MeasureLong TRANSPARENT_RETRIES_PER_CALL =
+          Measure.MeasureLong.create(
+              "grpc.io/client/transparent_retries_per_call", "Transparent retries per call", "1");
+      MeasureDouble RETRY_DELAY_PER_CALL =
+          Measure.MeasureDouble.create(
+              "grpc.io/client/retry_delay_per_call", "Retry delay per call", "ms");
+      MeasureMap measureMap = module.statsRecorder.newMeasureMap()
+          .put(RETRIES_PER_CALL, callAttemptTracer.retriesPerCall.get())
+          .put(TRANSPARENT_RETRIES_PER_CALL, callAttemptTracer.transparentRetriesPerCall.get())
+          .put(RETRY_DELAY_PER_CALL, callAttemptTracer.getRetryDelayNanos()/ NANOS_PER_MILLI);
+      if (!status.isOk()) {
+        measureMap.put(DeprecatedCensusConstants.RPC_CLIENT_ERROR_COUNT, 1);
+      }
+      TagValue methodTag = TagValue.create(fullMethodName);
+      TagValue statusTag = TagValue.create(status.getCode().toString());
+      measureMap.record(
+          module
+              .tagger
+              .toBuilder(parentCtx)
+              .putLocal(RpcMeasureConstants.GRPC_CLIENT_METHOD, methodTag)
+              .putLocal(RpcMeasureConstants.GRPC_CLIENT_STATUS, statusTag)
+              .build());
+    }
+  }
+
+  private static final class CallAttemptStatsTracer extends CallAttemptTracer {
+    private final CensusStatsModule module;
+    private final TagContext tagContext;
+    private final String fullMethodName;
+    private final Stopwatch stopwatch;
+
+    AtomicLong retriesPerCall;
+    AtomicLong transparentRetriesPerCall;
+    List<ClientCallTracer> streamTracerFactories;
+    List<Long> reactiveTimestamps;
+    List<Long> inactiveTimestamps;
+    AtomicLong activeStreams;
+
+    CallAttemptStatsTracer(
+        CensusStatsModule module, TagContext tagContext, String fullMethodName,
+        Stopwatch stopwatch) {
+      this.module = checkNotNull(module, "module");
+      this.tagContext = checkNotNull(tagContext, "tagContext");
+      this.fullMethodName = checkNotNull(fullMethodName, "fullMethodName");
+      this.stopwatch = stopwatch;
+    }
+
+    @Override
+    public ClientStreamTracer.Factory newAttempt(int attemptId, boolean transparentRetry) {
```

Comment:
@ejona86 If we use 

```java
ClientStreamTracer newClientStreamTracer(StreamInfo info, Metadata headers)
```

instead, how do we populate `boolean transparentRetry`? Using a special header?

Reply:
It'd be a new argument, and we may still need `attemptId` (unclear)

Reply:
Actually, I guess it could be added in StreamInfo.